### PR TITLE
Webapps tutorial

### PR DESCRIPTION
**Pull request by Naturalbornhippy**
Hi again,

i figured out how to test this. sorry for the first three untested requests. I thought if it is displayed it is enough. Now i have build it on my machine and this build runs. Hopefully it does now on the travis main repo too. If not i will stop my work and someone has to tell me how to do this.

For the request: 
I didnt found a real tutorial for webapp development (some threads on the forums, but nothing in the docs or wikis). Thats why i wrote this small howto. I hope this works for everyone. i will improve the page from time to time.

### CODE REVIEW

**Comment by UniversalSuperBox** [comment]
For future reference, if you push more commits to the branch that you select in your Pull Request, they'll automatically be added to the PR.

**Comment by UniversalSuperBox** [comment]
image

How do you feel about this addition to the layout, @NeoTheThird ?

File: appdev/webapp.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+alternate-webapp-generator
+--------------------------
+
+The alternate-webapp-generator will download all the stuff you need for a webapp and will add the the url, icon and a lot of meta-information to the app.
+In the end a .click package will be created, which can be uploaded to the OpenStore
+
+Download the following zip-file and unzip
+::
+
+    https://github.com/Winael/ubuntu/blob/master/ubuntuphone/webapps/Reboot/alternate-webapp-generator.zip
```
Comment (NeoTheThird) [inline appdev/webapp.rst]: Either make this a command, or a hyperlink. Just having the url isn't handy.

File: appdev/webapp.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    chmod +x /path/to/alternate-webapp-generator.sh
+
+Now we can start and edit the config.cfg.
+::
+
+    # config.cfg
+
+    export namespace="namespace"
+
```
Comment (NeoTheThird) [inline appdev/webapp.rst]: Do we want the whitespace here?

File: appdev/webapp.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+::
+
+    https://github.com/Winael/ubuntu/blob/master/ubuntuphone/webapps/Reboot/alternate-webapp-generator.zip
+
+Make the app-creator executable:
+::
+
+    chmod +x /path/to/alternate-webapp-generator.sh
+
+Now we can start and edit the config.cfg.
```
Comment (NeoTheThird) [inline appdev/webapp.rst]: What is a config.cfg, something ducks walk on? Jk, but this comes a little out of the blue. Maybe explain above that in the process we'll create a new file that holds all the information, called config.cfg.

**Comment by lduboeuf** [comment]
there is now `clickable init` that allow us to start with a webapp template, also `webapp creator` on openstore.

**Comment by lduboeuf** [comment]
maybe have also a reference or a summary of this: https://docs.ubuntu.com/phone/en/web/ubuntu-webapps-guide ?

File: appdev/webapp.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+::
+
+    # config.cfg
+
+    export namespace="namespace"
+    
+    export app_name="yourname"
```
Comment (NeoTheThird) [inline appdev/webapp.rst]: whycantthishavespaces? The namespace for the app-id is taken from the namespace field, so this should be formatted human readable "Your Name".

File: appdev/webapp.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    export app_title="YourTitle"
```
Comment (NeoTheThird) [inline appdev/webapp.rst]: This can have spaces as well

File: appdev/webapp.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+    ./alternate-webapp-generator.sh 
+
+This should now create a folder in the same directory which is named after the app_name attribute from the config.cfg
+In this folder you will find a file called: namespace.app_name_app_version_all.click
+
+Upload it to OpenStore
+----------------------
+
+This package can now be uploaded to  `OpenStore <https://open.uappexplorer.com/submit>`__. Just create an account an follow the instructions on the screen.
```
Comment (NeoTheThird) [inline appdev/webapp.rst]: You can't create an account on the openstore, but log in with ubuntuone or github.

**Comment by NeoTheThird** [comment]
Good points, @ldeboeuf! Using clickable would probably be a little cleaner, and it allows for cross-plattform instructions. This guide here will (i think) only work on 16.04.

Summing up the important parts of the document you linked to would also be a good idea, but of course without the ubuntu-sdk-ide stuff, since that is unsupported.

File: appdev/webapp.rst (hunk 3 above)
Comment (Naturalbornhippy) [reply appdev/webapp.rst]: Its included in the zip package - i will add a line with this information

File: appdev/webapp.rst (hunk 2 above)
Comment (Naturalbornhippy) [reply appdev/webapp.rst]: Its from the template file, but i can remove it from here - i dont mind

File: appdev/webapp.rst (hunk 5 above)
Comment (Naturalbornhippy) [reply appdev/webapp.rst]: Answer to the last two commands: actually both things would crash that script:
for example: 'ipes-ent.ipes ent' not properly formatted

File: appdev/webapp.rst (hunk 2 above)
Comment (NeoTheThird) [reply appdev/webapp.rst]: I meant the line breaks, actually, not the space. A space between the comment operator and the comment is generally considered a good practice, so i guess that should stay

File: appdev/webapp.rst [hunk 7]
```diff
@@ -13,17 +13,17 @@ In the end a .click package will be created, which can be uploaded to the OpenSt
 Download the following zip-file and unzip
 ::
 
-    https://github.com/Winael/ubuntu/blob/master/ubuntuphone/webapps/Reboot/alternate-webapp-generator.zip
+    wget https://raw.githubusercontent.com/Winael/ubuntu/master/ubuntuphone/webapps/Reboot/alternate-webapp-generator.zip
 
 Make the app-creator executable:
 ::
 
     chmod +x /path/to/alternate-webapp-generator.sh
```
Comment (NeoTheThird) [inline appdev/webapp.rst]: You could make this codeblock a little nicer if you also added the unzip command and joined them all into one block:
```
Download, extract and make the script executable::

    wget ...
    unzip...
    cd...
    chmod...

```

Comment (UniversalSuperBox) [reply appdev/webapp.rst]: Also, while technically correct to do it this way, I'd prefer that the two colons were at the end of the previous line.

```
Download the following zip-file and unzip::

    code

Make the app-creator executable::

    code
```

**Pull request closed without merging**